app/services/katello/event_daemon/runner.rb

```diff
@@ -0,0 +1,115 @@
+module Katello
+  module EventDaemon
+    class Runner
+      @services = {}
+      @cache = ActiveSupport::Cache::MemoryStore.new
+      @cache_keys = {}
+
+      class << self
+        def initialize
+          FileUtils.touch(lock_file)
+        end
+
+        def settings
+          SETTINGS[:katello][:event_daemon]
+        end
+
+        def pid
+          return unless pid_file && File.exist?(pid_file)
+
+          File.open(pid_file) { |f| f.read.to_i }
+        end
+
+        def pid_file
+          pid_dir.join('katello_event_daemon.pid')
+        end
+
+        def pid_dir
+          Rails.root.join('tmp', 'pids')
+        end
+
+        def lock_file
+          Rails.root.join('tmp', 'katello_event_daemon.lock')
+        end
+
+        def write_pid_file
+          return unless pid_file
+
+          FileUtils.mkdir_p(pid_dir)
+          File.open(pid_file, 'w') { |f| f.puts Process.pid }
+        end
+
+        def stop
+          return unless pid == Process.pid
+          Rails.logger.info "<<<<<<<<<<<<<<< STOPPING RUNNER >>>>>>>>>>>>>>>>>"
+          @monitor_thread.kill
+          @cache.clear
+          @services.values.each do |service_class|
+            service_class.close
+            Rails.logger.info "Clearing cache #{@cache_keys[service_class]}"
+            Rails.cache.delete(@cache_keys[service_class])
```

Comment:
How vital is it that we clear the cache here?

Reply:
I didn't really think about it, I was just moving the existing functionality over from the individual service classes.  But to answer the question, once the service is `close`d then our cached data is inaccurate anyway, so it makes sense to me to clear it.

Reply:
:heavy_check_mark: Solution